package/health_daemon/src/utils.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (C) 2018 Swift Navigation Inc.
+ * Contact: Swift Navigation <dev@swiftnav.com>
+ *
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+#include <libsbp/sbp.h>
+#include <string.h>
+
+#include "utils.h"
+
+static const char * const bool_names[] = {"False", "True"};
+static const char * const section_acquisition = SETTING_SECTION_ACQUISITION;
+static const char * const name_glonass_enabled = SETTING_GLONASS_ACQUISITION_ENABLED;
+
+int health_util_parse_setting_read_resp(const u8 *msg, u8 msg_n,
```

Comment:
I've seen this function re-implemented in at least 3 places in the code 😢

Reply:
I totally agree this is a tragedy. What I found was [here](https://github.com/swift-nav/piksi_buildroot/blob/c16da4843386031d1630e5e1d02c6dd18a06dec4/package/libpiksi/libpiksi/src/settings.c#L517) and [here](https://github.com/swift-nav/piksi_firmware_private/blob/6e7483e7636c7e35df55886a2edcc3824abbdab5/src/utils/settings/settings.c#L339), neither of which was something I could include nicely in source. And when I tried to copy paste them (bad I know) they didn't even work! I've created an issue for this https://github.com/swift-nav/firmware_team_planning/issues/446